Rename oh_action to oh_user_action for clarity

enyst opened the pull request:
This PR proposes to rename `oh_action` since it doesn't represent an OpenHands action, but a small subset of them: those _from the user_ using the web UI. Currently, the events passed via socket.io are `oh_event` and `oh_action`, where 'oh_action' is from the FE to BE.

`oh_action` seems a bit strange, since "actions" are _agent_ actions, _user_ actions as necessary of course, and maybe even _environment_ actions sometimes. The source of the action can be any of those.

I don't really know what name is best here, this is just a suggestion that maybe "oh_user_action" might work, even if there are some user actions that are not via web UI. The point is not to be perfect, it's to maybe not be too confusing to new contributors - and LLMs. 😅 

(I have to argue every time with the LLM when we touch this, it keeps trying to say that "actions are different than events" or that actions are "coming from the web socket", none of which... is great)

---

(openhands-ai)

## Description
This PR renames `oh_action` to `oh_user_action` across the codebase to improve clarity and self-documentation.

## Motivation
The rename is needed because "action" is an important category of "event" in the architecture, while "oh_action" specifically refers to user-initiated actions from frontend to backend. The current name creates confusion and does not self-document the origin of these events.

## Changes
- Renamed "oh_action" to "oh_user_action" in backend SocketIO event handlers
- Updated web UI event emitters to use the new name
- Updated documentation and schemas to reflect the change

## Benefits
- Improves code clarity by making the event name self-documenting
- Eliminates semantic confusion between general "actions" concept and user-initiated commands
- Explicitly indicates command origin in the event name

Fixes #66

rbren commented on openhands/server/listen_socket.py:
```diff
@@ -72,7 +72,7 @@ async def connect(connection_id: str, environ):
 
 @sio.event
-async def oh_action(connection_id: str, data: dict):
+async def oh_user_action(connection_id: str, data: dict):
```
Can we leave the old one here for now, with a TODO to remove? Otherwise launching this will break in-progress sessions (since the backend will reload but the FE will stay at the old version)

enyst replied:
@openhands Can you take care of this comment? We need both old and new for compatibility.

enyst commented:
I have added the old `oh_action` handler alongside the new `oh_user_action` handler for backward compatibility. This will ensure that both old and new clients can continue to work while the transition is happening. I also added a TODO comment to remind us to remove the deprecated handler once all clients have been updated to use the new event name.
